Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM bats/bats:latest@sha256:c707c5b7f9afd49da3e8a94248b03832e8b427f1b8f1ae9ec3cfdc5596d9c9f4
+FROM bats/bats:latest-no-faccessat2@sha256:ab8b147d6fd604a25872580db473a61d446370af00addfe0a0d8b4394eae0f6b
```

Comment:
Good idea! I'll double check and commit it in :)

Reply:
**Local Testing with `bk local run`**
```
➜  buildkite-plugin-tester git:(xmas-2022-updates) ✗ bk local run
>>> Starting local agent 🤖
>>> Starting build 👟
>>> Executing initial command: buildkite-agent pipeline upload
>>> Executing command step :docker: 🔨

~~~ Preparing plugins
# Plugin "github.com/buildkite-plugins/docker-compose-buildkite-plugin" will be checked out to "/var/folders/g5/gldb9rfd6xv_6s5fh1x4w3840000gp/T/buildkite-plugins-129606116/github-com-buildkite-plugins-docker-compose-buildkite-plugin-v4-9-0"
# Switching to the temporary plugin directory
$ cd /var/folders/g5/gldb9rfd6xv_6s5fh1x4w3840000gp/T/buildkite-plugins-129606116/github-com-buildkite-plugins-docker-compose-buildkite-plugin-v4-9-03961567702
$ git clone -v --recursive -- https://github.com/buildkite-plugins/docker-compose-buildkite-plugin .
Cloning into '.'...
POST git-upload-pack (193 bytes)
POST git-upload-pack (gzip 3820 to 1927 bytes)
remote: Enumerating objects: 3676, done.
remote: Counting objects: 100% (856/856), done.
remote: Compressing objects: 100% (341/341), done.
remote: Total 3676 (delta 575), reused 734 (delta 513), pack-reused 2820
Receiving objects: 100% (3676/3676), 901.79 KiB | 648.00 KiB/s, done.
Resolving deltas: 100% (2114/2114), done.
# Checking out `v4.9.0`
$ git checkout -f v4.9.0
Note: switching to 'v4.9.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at bf6b5a1 Merge pull request #364 from buildkite-plugins/toote_separator_issue-363
# Moving temporary plugin directory to final location
$ cd /private/var/folders/g5/gldb9rfd6xv_6s5fh1x4w3840000gp/T/buildkite-build-098799705
$ cd /Users/frank.peng/Repos/experiments/git-forks/buildkite-plugin-tester

~~~ Running plugin docker-compose command hook
$ /var/folders/g5/gldb9rfd6xv_6s5fh1x4w3840000gp/T/buildkite-plugins-129606116/github-com-buildkite-plugins-docker-compose-buildkite-plugin-v4-9-0/hooks/command
2023-01-04 15:22:10 WARN   POST http://127.0.0.1:55769/jobs/b1bbe304-2528-4310-8de4-2337551059db/data/exists: 404 (Attempt 1/10 Retrying in 5s)
2023-01-04 15:22:10 FATAL  Failed to see if meta-data exists: POST http://127.0.0.1:55769/jobs/b1bbe304-2528-4310-8de4-2337551059db/data/exists: 404

~~~ :docker: Building Docker Compose Service: tests
⚠️ No pre-built image found from a previous 'build' step for this service and config file. Building image...
$ docker-compose -f docker-compose.yml -p buildkiteb1bbe304252843108de42337551059db build --pull tests
[+] Building 1.8s (2/3)                                                         
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 2.25kB                                     0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for docker.io/bats/bats:1.8.2-no-faccessat2@  1.8s
Failed to fire hook: while creating logrus local file hook: user: Current requires cgo or $USER set in environment
[2023-01-04T04:22:13.151108000Z][docker-credential-desktop][F] user: Current requires cgo or $USER set in environment
[common/pkg/paths.Home()
[	common/pkg/paths/paths.go:105 +0x58
[common/pkg/paths.Container()
[	common/pkg/paths/user_darwin.go:37 +0x1c
[common/pkg/paths.Data()
[	common/pkg/paths/paths_darwin.go:27 +0x1c
[+] Building 2.3s (2/3)                                                         
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 2.25kB                                     0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for docker.io/bats/bats:1.8.2-no-faccessat2@  2.2s
Failed to fire hook: while creating logrus local file hook: user: Current requires cgo or $USER set in environment
[2023-01-04T04:22:13.626387000Z][docker-credential-desktop][F] user: Current requires cgo or $USER set in environment
[common/pkg/paths.Home()
[	common/pkg/paths/paths.go:105 +0x58
[common/pkg/paths.Container()
[	common/pkg/paths/user_darwin.go:37 +0x1c
[common/pkg/paths.Data()
[	common/pkg/paths/paths_darwin.go:27 +0x1c
[+] Building 2.4s (3/3) FINISHED                                                
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 2.25kB                                     0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => ERROR [internal] load metadata for docker.io/bats/bats:1.8.2-no-facce  2.4s
------
 > [internal] load metadata for docker.io/bats/bats:1.8.2-no-faccessat2@sha256:ed084f4b241c7e43422ff0a1d624a3a9609ef804ac8953449d2da63d6b8246e0:

------
failed to solve: rpc error: code = Unknown desc = failed to solve with frontend dockerfile.v0: failed to create LLB definition: rpc error: code = Unknown desc = error getting credentials - err: exit status 1, out: ``
🚨 Error: The command exited with status 17
^^^ +++
^^^ +++
user command error: The plugin docker-compose command hook exited with status 17

~~~ Running plugin docker-compose pre-exit hook
$ /var/folders/g5/gldb9rfd6xv_6s5fh1x4w3840000gp/T/buildkite-plugins-129606116/github-com-buildkite-plugins-docker-compose-buildkite-plugin-v4-9-0/hooks/pre-exit

~~~ :docker: Cleaning up after docker-compose
$ docker-compose -f docker-compose.yml -p buildkiteb1bbe304252843108de42337551059db kill
no container to kill
$ docker-compose -f docker-compose.yml -p buildkiteb1bbe304252843108de42337551059db rm --force -v
No stopped containers

$ docker-compose -f docker-compose.yml -p buildkiteb1bbe304252843108de42337551059db down --volumes
Warning: No resource found to remove for project "buildkiteb1bbe304252843108de42337551059db".


>>> 🚨 Command failed in 8.191923208s
🚨 Job failed with code 17
```

Seems like it's having trouble retrieving the metadata with the multi-architecture image sha

Reply:
**Local Testing with `bk local run`**
```
➜  buildkite-plugin-tester git:(xmas-2022-updates) ✗ bk local run
>>> Starting local agent 🤖
>>> Starting build 👟
>>> Executing initial command: buildkite-agent pipeline upload
>>> Executing command step :docker: 🔨

~~~ Preparing plugins
# Plugin "github.com/buildkite-plugins/docker-compose-buildkite-plugin" will be checked out to "/var/folders/g5/gldb9rfd6xv_6s5fh1x4w3840000gp/T/buildkite-plugins-478167075/github-com-buildkite-plugins-docker-compose-buildkite-plugin-v4-9-0"
# Switching to the temporary plugin directory
$ cd /var/folders/g5/gldb9rfd6xv_6s5fh1x4w3840000gp/T/buildkite-plugins-478167075/github-com-buildkite-plugins-docker-compose-buildkite-plugin-v4-9-03478990158
$ git clone -v --recursive -- https://github.com/buildkite-plugins/docker-compose-buildkite-plugin .
Cloning into '.'...
POST git-upload-pack (193 bytes)
POST git-upload-pack (gzip 3820 to 1927 bytes)
remote: Enumerating objects: 3676, done.
remote: Counting objects: 100% (856/856), done.
remote: Compressing objects: 100% (338/338), done.
remote: Total 3676 (delta 575), reused 742 (delta 516), pack-reused 2820
Receiving objects: 100% (3676/3676), 901.39 KiB | 1.11 MiB/s, done.
Resolving deltas: 100% (2115/2115), done.
# Checking out `v4.9.0`
$ git checkout -f v4.9.0
Note: switching to 'v4.9.0'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by switching back to a branch.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -c with the switch command. Example:

  git switch -c <new-branch-name>

Or undo this operation with:

  git switch -

Turn off this advice by setting config variable advice.detachedHead to false

HEAD is now at bf6b5a1 Merge pull request #364 from buildkite-plugins/toote_separator_issue-363
# Moving temporary plugin directory to final location
$ cd /private/var/folders/g5/gldb9rfd6xv_6s5fh1x4w3840000gp/T/buildkite-build-857260756
$ cd /Users/frank.peng/Repos/experiments/git-forks/buildkite-plugin-tester

~~~ Running plugin docker-compose command hook
$ /var/folders/g5/gldb9rfd6xv_6s5fh1x4w3840000gp/T/buildkite-plugins-478167075/github-com-buildkite-plugins-docker-compose-buildkite-plugin-v4-9-0/hooks/command
2023-01-04 15:18:09 WARN   POST http://127.0.0.1:55705/jobs/f9ff3962-8034-4ad0-ad6e-c5f0a41e899f/data/exists: 404 (Attempt 1/10 Retrying in 5s)
2023-01-04 15:18:09 FATAL  Failed to see if meta-data exists: POST http://127.0.0.1:55705/jobs/f9ff3962-8034-4ad0-ad6e-c5f0a41e899f/data/exists: 404

~~~ :docker: Building Docker Compose Service: tests
⚠️ No pre-built image found from a previous 'build' step for this service and config file. Building image...
$ docker-compose -f docker-compose.yml -p buildkitef9ff396280344ad0ad6ec5f0a41e899f build --pull tests
[+] Building 0.9s (2/3)                                                         
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 2.25kB                                     0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for docker.io/bats/bats:1.8.2-no-faccessat2@  0.9s
Failed to fire hook: while creating logrus local file hook: user: Current requires cgo or $USER set in environment
[2023-01-04T04:18:11.226801000Z][docker-credential-desktop][F] user: Current requires cgo or $USER set in environment
[common/pkg/paths.Home()
[	common/pkg/paths/paths.go:105 +0x58
[common/pkg/paths.Container()
[	common/pkg/paths/user_darwin.go:37 +0x1c
[common/pkg/paths.Data()
[	common/pkg/paths/paths_darwin.go:27 +0x1c
[+] Building 1.4s (2/3)                                                         
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 2.25kB                                     0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for docker.io/bats/bats:1.8.2-no-faccessat2@  1.3s
Failed to fire hook: while creating logrus local file hook: user: Current requires cgo or $USER set in environment
[2023-01-04T04:18:11.646996000Z][docker-credential-desktop][F] user: Current requires cgo or $USER set in environment
[common/pkg/paths.Home()
[	common/pkg/paths/paths.go:105 +0x58
[common/pkg/paths.Container()
[	common/pkg/paths/user_darwin.go:37 +0x1c
[common/pkg/paths.Data()
[	common/pkg/paths/paths_darwin.go:27 +0x1c
[+] Building 6.6s (12/12) FINISHED                                              
 => [internal] load build definition from Dockerfile                       0.0s
 => => transferring dockerfile: 2.25kB                                     0.0s
 => [internal] load .dockerignore                                          0.0s
 => => transferring context: 2B                                            0.0s
 => [internal] load metadata for docker.io/bats/bats:1.8.2-no-faccessat2@  1.4s
 => [1/8] FROM docker.io/bats/bats:1.8.2-no-faccessat2@sha256:ab8b147d6fd  0.0s
 => CACHED [2/8] RUN apk --no-cache add ncurses curl jq                    0.0s
 => CACHED [3/8] RUN mkdir -p /usr/local/lib/bats/bats-support     && cur  0.0s
 => CACHED [4/8] RUN mkdir -p /usr/local/lib/bats/bats-assert     && curl  0.0s
 => [5/8] RUN mkdir -p /usr/local/lib/bats/bats-mock     && curl -sSL htt  3.3s
 => [6/8] RUN mkdir -p /usr/local/lib/bats/bats-file      && curl -sSL ht  1.5s
 => [7/8] RUN if [[ -e /bin/bash ]]; then echo "/bin/bash already exists"  0.3s
 => [8/8] WORKDIR /plugin                                                  0.0s
 => exporting to image                                                     0.0s
 => => exporting layers                                                    0.0s
 => => writing image sha256:ef762508a8f14fa20f46fcc1e1bd58d5e0c0231e008c5  0.0s
 => => naming to docker.io/library/buildkitef9ff396280344ad0ad6ec5f0a41e8  0.0s

Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them


~~~ :docker: Starting dependencies
$ docker-compose -f docker-compose.yml -p buildkitef9ff396280344ad0ad6ec5f0a41e899f up -d --scale tests=0 tests
[+] Running 1/0
 ⠿ Network buildkitef9ff396280344ad0ad6ec5f0a41e899f_default  Created 0.0s


+++ :docker: Running  in service tests
$ docker-compose -f docker-compose.yml -p buildkitef9ff396280344ad0ad6ec5f0a41e899f run --name buildkitef9ff396280344ad0ad6ec5f0a41e899f_tests_build_1 --rm tests
test-failure.bats
 ✓ bats intentional failure - expects Not AOK
 ✓ bats-mock should throw error
 ✓ bats intentional failure - alerts error
test-success.bats
 ✓ bats works AOK
 ✓ bats-mock works

5 tests, 0 failures



~~~ Running plugin docker-compose pre-exit hook
$ /var/folders/g5/gldb9rfd6xv_6s5fh1x4w3840000gp/T/buildkite-plugins-478167075/github-com-buildkite-plugins-docker-compose-buildkite-plugin-v4-9-0/hooks/pre-exit

~~~ :docker: Cleaning up after docker-compose
$ docker-compose -f docker-compose.yml -p buildkitef9ff396280344ad0ad6ec5f0a41e899f kill
no container to kill
$ docker-compose -f docker-compose.yml -p buildkitef9ff396280344ad0ad6ec5f0a41e899f rm --force -v
No stopped containers

$ docker-compose -f docker-compose.yml -p buildkitef9ff396280344ad0ad6ec5f0a41e899f down --volumes
[+] Running 1/0
 ⠿ Network buildkitef9ff396280344ad0ad6ec5f0a41e899f_default  Removed 0.1s


>>> Command succeeded in 19.121200958s
>>> Build finished in 19.121254208s
```


On the other hand, the linux/amd64 arch image sha works fine.

Reply:
Is this just a limitation of the `bk cli` tool or am I missing something here?

Reply:
After some research into the error, it seems like an issue that is caused by Docker Buildkit. I've tried using the `buildkit: true` and `cli-version: 2` option for `docker-compose-buildkite-plugin`, but it doesn't seem to work. 

I've also messed around with the `daemon.json` locally for my Docker Engine, to enable/disable the buildkit, but that hasn't worked either. 

Otherwise, as a last resort, I might just revert back to `linux/amd64` arch image instead of `multi` arch image, but pinned to version `1.8.2-no-faccessat2`.

Locally, if I do everything via `docker build` or `docker buildx build`, it seems to build the image successfully. 

I can't help but just feel like I'm missing something. 

Any suggestions? @toote

Reply:
That is indeed very weird as I can not reproduce. But I am running on Linux and - based on what I'm reading - you are running on a Mac.

From the logs, it would appear that the actual issue is with the configuration of the credentials helper:

```
failed to create LLB definition: rpc error: code = Unknown desc = error getting credentials - err: exit status 1, out: ``
```

Reply:
Yep, running on macOS Ventura 13.1 (Darwin Kernel Version 22.2.0).